<launch>

  <arg name="max_speed" default="0.4" /> 
  <arg name="max_turning_speed" default="2.0" /> 
  <arg name="distance_tolerance" default="0.15" /> 
  <arg name="align_tolerance_deg" default="5.0" /> 
  <arg name="control_rate" default="60" />
  <arg name="waypoint_dir" default="$(find husky_waypoint_nav)/config/waypoints" />
  <arg name="waypoint_prefix" default="optitrack_waypoints_" />
  <arg name="log_dir" default="$(find husky_waypoint_nav)/logs" /> <arg name="pose_topic" default="/natnet_ros/base_link/pose" /> <arg name="Kp_angular" default="0.25" /> <arg name="Kd_angular" default="0.1" /> <arg name="Kp_linear" default="0.5" /> <node pkg="husky_waypoint_nav" type="pd_navigator.py" name="pd_ads_navigator" output="screen" respawn="false">
    <param name="controller_type" value="PD_Align_Drive_Straight"/>

    <param name="max_speed" value="$(arg max_speed)"/>
    <param name="max_turning_speed" value="$(arg max_turning_speed)"/>
    <param name="distance_tolerance" value="$(arg distance_tolerance)"/>
    <param name="align_tolerance_deg" value="$(arg align_tolerance_deg)"/>
    <param name="control_rate" value="$(arg control_rate)"/>
    <param name="waypoint_dir" value="$(arg waypoint_dir)" />
    <param name="waypoint_prefix" value="$(arg waypoint_prefix)" />
    <param name="log_dir" value="$(arg log_dir)" />
    <param name="pose_topic" value="$(arg pose_topic)" />

    <param name="Kp_angular" value="$(arg Kp_angular)"/>
    <param name="Kd_angular" value="$(arg Kd_angular)"/>
    <param name="Kp_linear" value="$(arg Kp_linear)"/> </node>

</launch>